### PR TITLE
chore: Add some :sparkles: to Typescript conversion 

### PR DESCRIPTION
**Pull request by erikshestopal**
Extending some of Alex's awesome work to add stricter type safety and use some different approaches to reduce boilerplate.

### CODE REVIEW

File: src/createTheme.spec.ts [hunk 1]
```diff
@@ -16,7 +17,7 @@ describe('createTheme', () => {
         ...defaultThemeColors,
         primaryDark: 'foo',
       },
-    });
+    } as const);
 
     expect(theme.colors.primaryDark).toEqual('foo');
```
Comment (vietnogigoodrx) [inline src/createTheme.spec.ts:22]: @erikshestopal Do you get types for `theme.colors.primaryDark` ? My VsCode is treating it as `any`.

Comment (erikshestopal) [reply src/createTheme.spec.ts:22]: I had to restart the TS server 
Screen Shot 2020-07-27 at 12 08 20 PM

File: src/createTheme.spec.ts [hunk 2]
```diff
@@ -16,7 +17,7 @@ describe('createTheme', () => {
         ...defaultThemeColors,
         primaryDark: 'foo',
```
Comment (vietnogigoodrx) [inline src/createTheme.spec.ts:18]: @erikshestopal  I dont get an error if typed the key wrong, `primaryDark` -> `priimaryDark`

Comment (erikshestopal) [reply src/createTheme.spec.ts:18]: Screen Shot 2020-07-27 at 12 20 24 PM

Comment (erikshestopal) [reply src/createTheme.spec.ts:18]: Or are you saying if you add the typo to the override `colors` object?

Comment (vietnogigoodrx) [reply src/createTheme.spec.ts:18]: I meant online 18

Comment (erikshestopal) [reply src/createTheme.spec.ts:18]: @vietnogigoodrx do you think you can fix that issue? I hacked this over the weekend and didn't address small edge cases. I have some other priorities that I have to work on but would love to help sometime later this week.

**Pull request closed without merging**